mars-sim-core/src/main/java/com/mars_sim/core/mission/MissionStep.java

```diff
@@ -99,28 +98,28 @@ protected void addLifeSupportResource(int crew, double durationMSol, boolean ide
      * @param task Task allocated
      */
     protected boolean assignTask(Worker worker, Task task) {
```

Comment:
This assignTask() method is not being used at this moment.

The two real assignTask() methods (one for Person, one for Robot) that are in use are in AbstractMission. They are more refined than the old one here.

@bevans2000 
As a side note, personally speaking, I don't like the approach of first creating an instance of a task and then inserting it here as a parameter into assignTask() method. 

Many CPU cycles would have been wasted if the precondition for this task is not met and we would have to cancel this task.

Reply:
MissionDisembarkStep does

Reply:
Sure but tat is a massive redesign of how Tasks work. Lets focus on achievable changes.